CONTRIBUTING.md

```diff
@@ -1,25 +1,70 @@
-# Issues
+# Contributing
 
-Issues have three types: bug, feature request, and support. When reporting a bug, you must include
-the operating system used, any relevant information about the tech stack, and the feature flags
-used, as specified in the issue template. Feature requests also have an issue template, containing
-questions that should be answered in the issue. We aim for 100% coverage of the Discord API; we will
-wait until a new feature is supported in the [Discord API documentation] before adding its support
-to Twilight. Before making an issue, be sure to consider joining the [Twilight Discord] and bringing
-up your topic in the `#support` channel.
+Thank you for considering adding your contribution to Twilight! This project
+would not be what it is without the support of the community. This document
+contains rules and guidelines that should be followed when making contributions
+to Twilight. 
 
-# Errors
+## Project Management
 
-Twilight's `Error` system is a struct with one required field (`kind`) and one optional field
-(`source`). It includes three methods (`kind`, `into_source`, and `into_parts`) which allow the user
-to access data within the error. The return types for any source errors are `dyn Error`, which
-allows us to update dependencies without breaking the public API.
+### Issues
 
-Normally, the fields of the error struct are not public, as they can be accessed through the
-provided methods. However, in some cases, you may need to make the struct fields `pub` or
-`pub(crate)`.
+There are two major kinds of issues in Twilight, user-created and
+developer-created. Of the user-created issues, they are usually bug reports,
+feature requests, or support requests. If reporting a bug, please include the
+operating system used, any relevant information about the tech stack, and the
+feature flags used. Before making a support issue, be sure to consider joining
+the [Twilight Discord] and making a thread for your question in the `#support`
+channel.
 
-Any new error implementations must follow this pattern:
+Developer-created issues are most often bugs or tracking issues for supporting
+Discord API features. We aim for 100% coverage of the Discord API; we will wait
+until a new feature is supported in the [Discord API] before adding its support
+to Twilight.
+
+### Pull Requests
+
+In order to ease developer cooperation, please make a pull request from a
+feature branch on your own fork instead of from `main` or `next`. Additionally,
+since we squash every commit before merging, force-pushing a branch with new
+changes or updates is not always required. This helps with maintaining a
+chronological history in the PR story.
+
+Contributors must add tests and documentation that reflects their changes.
+
+Twilight follows a [Conventional Commit] style for pull request titles and
+commits. A Github Actions workflow automatically labels pull requests based on
+parsing the title. For more information, see [this discussion][#1439].
```

Comment:
This document might better serve as a single source of truth. Can we bring any missing information here?